source/backend/dal/Repositories/ProjectRepository.cs

```diff
@@ -69,6 +69,36 @@ public Task<Paged<PimsProject>> GetPageAsync(ProjectFilter filter)
             return GetPage(filter);
         }
 
+        /// <summary>
+        /// Get by ID - Search Projects by Id.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public async Task<PimsProject> Get(long id)
+        {
+            User.ThrowIfNotAuthorized(Permissions.PropertyView);
+
+            return await Context.PimsProjects
+                    .AsNoTracking()
+                    .Include(x => x.ProjectStatusTypeCodeNavigation)
+                    .Where(x => x.Id == id)
+                    .FirstOrDefaultAsync();
+        }
+
+        /// <summary>
+        /// Add Project to Context.
+        /// </summary>
+        /// <param name="project"></param>
+        /// <returns></returns>
+        public async Task<PimsProject> Add(PimsProject project)
+        {
+            User.ThrowIfNotAuthorized(Permissions.ProjectAdd);
+
+            await Context.PimsProjects.AddAsync(project);
+            Context.CommitTransaction();
```

Comment:
Ideally, the service would have control over when to commit the transaction

Reply:
Updated